src/Lib/Router.php

```diff
@@ -168,11 +171,14 @@ public static function parseURI($uri)
      */
     public static function filterURI($uri)
     {
-        return preg_replace(
-            "/^(" . self::escape(self::$webdir) . "|" . self::escape(dirname(self::$webdir)) . "|\/)/i",
-            "",
-            $uri,
-            1
+        return rtrim(
+            preg_replace(
+                "/^(" . self::escape(self::$webdir) . "|" . self::escape(dirname(self::$webdir)) . "|\/)/i",
+                "",
+                str_replace('/?', '?', $uri),
```

Comment:
Is there a purpose for this str_replace?  It causes some of the unit tests to fail